lib/ddtrace/contrib/sidekiq/server_tracer.rb

```diff
@@ -0,0 +1,50 @@
+require 'ddtrace/contrib/sidekiq/tracing'
+
+module Datadog
+  module Contrib
+    module Sidekiq
+      # Tracer is a Sidekiq server-side middleware which traces executed jobs
+      class ServerTracer
+        include Tracing
+
+        def initialize(options = {})
+          super
+          @sidekiq_service = options[:service_name] || Datadog.configuration[:sidekiq][:service_name]
+        end
+
+        def call(worker, job, queue)
+          resource = job_resource(job)
+
+          service = service_from_worker_config(resource) || @sidekiq_service
+          set_service_info(service)
+
+          @tracer.trace(Ext::SPAN_JOB, service: service, span_type: Datadog::Ext::AppTypes::WORKER) do |span|
+            span.resource = resource
+            span.set_tag(Ext::TAG_JOB_ID, job['jid'])
+            span.set_tag(Ext::TAG_JOB_RETRY, job['retry'])
+            span.set_tag(Ext::TAG_JOB_QUEUE, job['queue'])
+            span.set_tag(Ext::TAG_JOB_WRAPPER, job['class']) if job['wrapped']
+            span.set_tag(Ext::TAG_JOB_DELAY, 1000.0 * (Time.now.utc.to_f - job['enqueued_at'].to_f))
+
+            yield
+          end
+        end
+
+        private
+
+        def service_from_worker_config(resource)
+          # Try to get the Ruby class from the resource name.
+          worker_klass = begin
+            Object.const_get(resource)
```

Comment:
This is also old code but... can you think of any scenarios where the resource (i.e. class name) would not exist in the namespace?

If failing to resolve a constant and raising an error is a very common case, then this could be incurring a significant performance cost. If that is true, we'd want to fix this (now or at some point in the future.) Otherwise if this is just an edge case, and a safety of sorts, then I don't think we need to worry about this.

This isn't a blocker for this PR most likely, since it's existing behavior.

Reply:
@delner: This is adapted from the original tracer implementation, but given your comment above I think this seems like a good enough time as any to remove support for workers defining a `.datadog_tracer_config` method (and therefore not need this `Object.const_get`). What do you think?

Reply:
Yeah I think we'll want to remove this too, but to do so, we'll have to deprecate the feature and remove it in a future release. I've added an item to our backlog to do each of these things in future PRs.